### ISSUE TITLE
[bug]: Firewall needs to be turned off manually to generate LetsEncrypt certificate.

### ISSUE THREAD

**Issue by ctagithub** [FreePBX/issue-tracker#77]
### FreePBX Version

FreePBX 17

### Issue Description

Admin, Certificate Management, New Certificate
Generate Lets Encrypt Certificate
Fails unless firewall is turned off.

If you manually disable firewall and generate cert it works fine.  Note that we have two networks, one internal, one external.  Firewall was tested new and tested with restore from FreePBX 16.

### Operating Environment

Two network connections, one internal NAT one external public dedicated IP.

Linux version 6.1.0-18-amd64 (debian-kernel@lists.debian.org) (gcc-12 (Debian 12.2.0-14) 12.2.0, GNU ld (GNU Binutils for Debian) 2.40) #1 SMP PREEMPT_DYNAMIC Debian 6.1.76-1 (2024-02-01)

+-------------------+------------+-------------------------------+-------------+-----------+
| Module            | Version    | Status                        | License     | Signature |
+-------------------+------------+-------------------------------+-------------+-----------+
| amd               | 17.0.1     | Enabled and up to date        | GPLv3+      | Sangoma   |
| announcement      | 17.0.2.1   | Enabled and up to date        | GPLv3+      | Sangoma   |
| areminder         | 17.0.3.9   | Enabled and up to date        | Commercial  | Sangoma   |
| arimanager        | 17.0.1     | Enabled and up to date        | GPLv3+      | Sangoma   |
| asterisk-cli      | 17.0.2     | Enabled and up to date        | GPLv3+      | Sangoma   |
| asteriskinfo      | 17.0.1     | Enabled and up to date        | GPLv3+      | Sangoma   |
| backup            | 17.0.5.14  | Enabled and up to date        | GPLv3+      | Sangoma   |
| blacklist         | 17.0.1.2   | Enabled and up to date        | GPLv3+      | Sangoma   |
| broadcast         | 17.0.1.6   | Enabled and up to date        | Commercial  | Sangoma   |
| builtin           |            | Enabled; Not available online |             | Unsigned  |
| bulkhandler       | 17.0.4     | Enabled and up to date        | GPLv3+      | Sangoma   |
| calendar          | 17.0.4.6   | Enabled and up to date        | GPLv3+      | Sangoma   |
| callaccounting    | 17.0.4     | Enabled and up to date        | Commercial+ | Sangoma   |
| callback          | 17.0.2.1   | Enabled and up to date        | GPLv3+      | Sangoma   |
| callerid          | 17.0.1     | Enabled and up to date        | Commercial  | Sangoma   |
| callforward       | 17.0.1.3   | Enabled and up to date        | AGPLv3+     | Sangoma   |
| calllimit         | 17.0.1.2   | Enabled and up to date        | Commercial  | Sangoma   |
| callrecording     | 17.0.3.6   | Enabled and up to date        | AGPLv3+     | Sangoma   |
| callwaiting       | 17.0.3.2   | Enabled and up to date        | GPLv3+      | Sangoma   |
| cdr               | 17.0.4.13  | Enabled and up to date        | GPLv3+      | Sangoma   |
| cel               | 17.0.2.7   | Enabled and up to date        | GPLv3+      | Sangoma   |
| certman           | 17.0.3.7   | Enabled and up to date        | AGPLv3+     | Sangoma   |
| cidlookup         | 17.0.1.1   | Enabled and up to date        | GPLv3+      | Sangoma   |
| conferences       | 17.0.3.1   | Enabled and up to date        | GPLv3+      | Sangoma   |
| conferencespro    | 17.0.1.6   | Enabled and up to date        | Commercial  | Sangoma   |
| configedit        | 17.0.1.1   | Enabled and up to date        | AGPLv3+     | Sangoma   |
| contactmanager    | 17.0.5.6   | Enabled and up to date        | GPLv3+      | Sangoma   |
| core              | 17.0.9.33  | Enabled and up to date        | GPLv3+      | Sangoma   |
| cos               | 17.0.1.1   | Enabled and up to date        | Commercial  | Sangoma   |
| customappsreg     | 17.0.1     | Enabled and up to date        | GPLv3+      | Sangoma   |
| dashboard         | 17.0.4.2   | Enabled and up to date        | AGPLv3+     | Sangoma   |
| daynight          | 17.0.1.1   | Enabled and up to date        | GPLv3+      | Sangoma   |
| dictate           | 17.0.1     | Enabled and up to date        | GPLv3+      | Sangoma   |
| directory         | 17.0.1.1   | Enabled and up to date        | GPLv3+      | Sangoma   |
| donotdisturb      | 17.0.2.1   | Enabled and up to date        | GPLv3+      | Sangoma   |
| endpoint          | 17.0.1.53  | Enabled and up to date        | Commercial  | Sangoma   |
| extensionroutes   | 17.0.1     | Enabled and up to date        | Commercial  | Sangoma   |
| extensionsettings | 17.0.1     | Enabled and up to date        | GPLv3+      | Sangoma   |
| fax               | 17.0.3.2   | Enabled and up to date        | GPLv3+      | Sangoma   |
| faxpro            | 17.0.1.14  | Enabled and up to date        | Commercial  | Sangoma   |
| featurecodeadmin  | 17.0.2     | Enabled and up to date        | GPLv3+      | Sangoma   |
| filestore         | 17.0.2.9   | Enabled and up to date        | AGPLv3      | Sangoma   |
| findmefollow      | 17.0.4.6   | Enabled and up to date        | GPLv3+      | Sangoma   |
| firewall          | 17.0.1.13  | Enabled and up to date        | AGPLv3+     | Sangoma   |
| framework         | 17.0.15.13 | Enabled and up to date        | GPLv2+      | Sangoma   |
| hotelwakeup       | 17.0.1.5   | Enabled and up to date        | GPLv2       | Sangoma   |
| iaxsettings       | 17.0.1     | Enabled and up to date        | AGPLv3      | Sangoma   |
| infoservices      | 17.0.1     | Enabled and up to date        | GPLv2+      | Sangoma   |
| ivr               | 17.0.5     | Enabled and up to date        | GPLv3+      | Sangoma   |
| languages         | 17.0.1     | Enabled and up to date        | GPLv3+      | Sangoma   |
| logfiles          | 17.0.2     | Enabled and up to date        | GPLv3+      | Sangoma   |
| manager           | 17.0.3     | Enabled and up to date        | GPLv2+      | Sangoma   |
| miscapps          | 17.0.3     | Enabled and up to date        | GPLv3+      | Sangoma   |
| miscdests         | 17.0.1.1   | Enabled and up to date        | GPLv3+      | Sangoma   |
| music             | 17.0.4     | Enabled and up to date        | GPLv3+      | Sangoma   |
| paging            | 17.0.3     | Enabled and up to date        | GPLv3+      | Sangoma   |
| pagingpro         | 17.0.1.6   | Enabled and up to date        | Commercial  | Sangoma   |
| parking           | 17.0.2     | Enabled and up to date        | GPLv3+      | Sangoma   |
| parkpro           | 17.0.1.4   | Enabled and up to date        | Commercial  | Sangoma   |
| pbxmfa            | 17.0.2     | Enabled and up to date        | Commercial+ | Sangoma   |
| pinsets           | 17.0.3     | Enabled and up to date        | GPLv3+      | Sangoma   |
| pinsetspro        | 17.0.2     | Enabled and up to date        | Commercial  | Sangoma   |
| pm2               | 17.0.3.1   | Enabled and up to date        | AGPLv3+     | Sangoma   |
| presencestate     | 17.0.2.1   | Enabled and up to date        | GPLv3+      | Sangoma   |
| printextensions   | 17.0.1.2   | Enabled and up to date        | GPLv3+      | Sangoma   |
| queueprio         | 17.0.1.4   | Enabled and up to date        | GPLv3+      | Sangoma   |
| queues            | 17.0.1.7   | Enabled and up to date        | GPLv2+      | Sangoma   |
| queuestats        | 17.0.1.6   | Enabled and up to date        | Commercial  | Sangoma   |
| qxact_reports     | 17.0.2     | Enabled and up to date        | Commercial  | Sangoma   |
| recording_report  | 17.0.3.6   | Enabled and up to date        | Commercial  | Sangoma   |
| recordings        | 17.0.2.2   | Enabled and up to date        | GPLv3+      | Sangoma   |
| restapps          | 17.0.1.14  | Enabled and up to date        | Commercial  | Sangoma   |
| ringgroups        | 17.0.2.4   | Enabled and up to date        | GPLv3+      | Sangoma   |
| sangomaconnect    | 17.0.1.20  | Enabled and up to date        | Commercial  | Sangoma   |
| sangomartapi      | 17.0.1.14  | Enabled and up to date        | Commercial  | Sangoma   |
| setcid            | 17.0.1.2   | Enabled and up to date        | GPLv3+      | Sangoma   |
| sipsettings       | 17.0.6.3   | Enabled and up to date        | AGPLv3+     | Sangoma   |
| sipstation        | 17.0.3.3   | Enabled and up to date        | Commercial  | Sangoma   |
| sms               | 17.0.1.8   | Enabled and up to date        | Commercial  | Sangoma   |
| smsplus           | 17.0.3     | Enabled and up to date        | Commercial  | Sangoma   |
| soundlang         | 17.0.4     | Enabled and up to date        | GPLv3+      | Sangoma   |
| sysadmin          | 17.0.1.48  | Enabled and up to date        | Commercial  | Sangoma   |
| timeconditions    | 17.0.1.13  | Enabled and up to date        | GPLv3+      | Sangoma   |
| tts               | 17.0.1.1   | Enabled and up to date        | GPLv3+      | Sangoma   |
| ttsengines        | 17.0.1     | Enabled and up to date        | AGPLv3      | Sangoma   |
| ucp               | 17.0.4.13  | Enabled and up to date        | AGPLv3+     | Sangoma   |
| userman           | 17.0.6.16  | Enabled and up to date        | AGPLv3+     | Sangoma   |
| vmblast           | 17.0.1     | Enabled and up to date        | GPLv3+      | Sangoma   |
| vmnotify          | 17.0.1.5   | Enabled and up to date        | Commercial  | Sangoma   |
| voicemail         | 17.0.5.14  | Enabled and up to date        | GPLv3+      | Sangoma   |
| voicemail_report  | 17.0.1.1   | Enabled and up to date        | Commercial  | Sangoma   |
| vqplus            | 17.0.1.13  | Enabled and up to date        | Commercial  | Sangoma   |
| weakpasswords     | 17.0.1     | Enabled and up to date        | GPLv3+      | Sangoma   |
| webrtc            | 17.0.2.1   | Enabled and up to date        | GPLv3+      | Sangoma   |
+-------------------+------------+-------------------------------+-------------+-----------+

### Relevant log output

```shell
Processing: hidden, Local IP: hidden, Public IP: dns error
Self test: trying http://hidden.com/.freepbx-known/7c8e49f694957e4a60db137ce5bfd79b
Self test error: Pest_Curl_Exec - Connection timed out after 5001 milliseconds
```

**Comment by jissphilip** [comment]
@ctagithub 

Please make sure all LE related settings are ON on firewall 

![image](https://github.com/FreePBX/issue-tracker/assets/30717824/7bafd87e-072f-48c5-8c74-c7497e96bcd0)

**Comment by ctagithub** [comment]
We reviewed our settings and this has been on. We went ahead and tested again with no luck. We do have two network connections, so maybe it is something with that?

**Comment by nobe80** [comment]
We have the same issue here. fresh debian 12 installation with FreePBX V17
We have also Processing: hidden, Local IP: 127.0.0.1, Public IP: dns error

But if we disable the firewall we also cannot generate LE certificates
@ctagithub how to manually disable the firewall? so via cli ?

**Comment by ramarajan222** [comment]
Hi @nobe80 looks like some DNS issue, Can you remove the hostname from /etc/hosts and then give it a try again?
Also you can stop the firewall using `fwconsole firewall stop`

**Comment by nobe80** [comment]
then we get:
LetsEncrypt Generation Failure
Self test error: Pest_Curl_Exec - Failed to connect to bitpbx.(oursecondleveldomain).de port 80 after 0 ms: Couldn't connect to server

    Does DNS for bitpbx.(oursecondleveldomain).de resolve correctly? Local DNS result: 127.0.0.1, External DNS result: dns error

Processing: bitpbx.(oursecondleveldomain).de, Local IP: 127.0.0.1, Public IP: dns error
Self test: trying http://bitpbx.(oursecondleveldomain).de/.freepbx-known/9a01978cf85fe24bfa4afbe0b80a5522
Self test error: Pest_Curl_Exec - Failed to connect to bitpbx.(oursecondleveldomain).de port 80 after 0 ms: Couldn't connect to server

**Comment by ramarajan222** [comment]
@nobe80  Are you able to access your PBX with hostname "bitpbx.oursecondleveldomain.de" outside of your network?

**Comment by ramarajan222** [comment]
hi @nobe80  looks like the home name is not resolving ` nslookup bitpbx.oursecondleveldomain.de
Server:         108.61.10.10
Address:        108.61.10.10#53

** server can't find bitpbx.oursecondleveldomain.de: NXDOMAIN
`

**Comment by nobe80** [comment]
Hi @ramarajan222 
yes perfectly. The Freepbx run in a datacenter with public ip here in germany and we have configured the DNS Name correctly. But we cannot create LE certificate :-/

Sorry! The oursecondleveldomain is simply a placeholder here!
Here the output with our correct domain:

LetsEncrypt Generation Failure
Self test error: Pest_Curl_Exec - Failed to connect to bitpbx.beonit-cloud.de port 80 after 0 ms: Couldn't connect to server

Does DNS for bitpbx.beonit-cloud.de resolve correctly? Local DNS result: 127.0.0.1, External DNS result: dns error

Processing: bitpbx.beonit-cloud.de, Local IP: 127.0.0.1, Public IP: dns error
Self test: trying http://bitpbx.beonit-cloud.de/.freepbx-known/9a01978cf85fe24bfa4afbe0b80a5522
Self test error: Pest_Curl_Exec - Failed to connect to bitpbx.beonit-cloud.de port 80 after 0 ms: Couldn't connect to server

```

**Comment by nobe80** [comment]
Not possible to generate LE certificate. Get always the same issue:

Verification timed out

Processing: bitpbx.beonit-cloud.de, Local IP: 116.203.228.6, Public IP: dns error
Self test: trying http://bitpbx.beonit-cloud.de/.freepbx-known/4cabc47582e507a090185e27bfb474a2
Self test: received 4cabc47582e507a090185e27bfb474a2

**Comment by ctagithub** [comment]
This issue is the same as ours, we get the same error.  I opened this issue.  How can it be a DNS issue if it works when you disable the firewall?  LE is trying to contact the server to approve the request over port 80, my understanding is that when you try and generate or the system auto updates the cert that it temp opens the firewall for LE to talk back then closes it again.  Confirmed that if you disable firewall you can gen cert and it will update cert.  In my case we have 2 network connections.  One internal and one external.  I believe this to work in single nic environment, seems to be a problem with dual, virtual net, more than one network in Debian.

**Comment by nobe80** [comment]
Jep i can also not imagine that is has something to do with DNS. Our DNS works fine but at our server we have only 1 nic adapter that is directly connected to the public internet.
So would be a solution to install LE certificate by another way? Like snap with certbot?

**Comment by ramarajan222** [comment]
Hi @nobe80 ,

The issue with Let's Encrypt certificate generation has been resolved with the latest firewall module (17.0.1.17). Please upgrade the firewall module and try again. 
 
Moving this jira to close, reopen if needed. thanks.

**Comment by ctagithub** [comment]
We verified that this issue is resolved now.  There was an odd problem where when we updated the module the network interfaces both showed as trusted which opened the external interface to the world so you might make sure that isn't something that happens with the update as that is bad.  After update we can update the cert and we can get a new cert with the firewall up.
